server/blob/blob-storage-strategy/src/main/java/org/apache/james/server/blob/deduplication/MinIOGenerationAwareBlobId.java

```diff
@@ -109,7 +119,18 @@ public String asString() {
         if (family == NO_FAMILY) {
             return delegate.asString();
         }
-        return family + "/" + generation + "/" + delegate.asString();
+
+        String blobIdPartAsString = delegate.asString();
+        if (blobIdPartAsString.length() < PATH_DELIMITED_NUMBER) {
+            return family + PATH_DELIMITER_CHARACTER + generation + PATH_DELIMITER_CHARACTER + blobIdPartAsString;
+        }
+
+        return family + PATH_DELIMITER_CHARACTER + generation + PATH_DELIMITER_CHARACTER +
+            blobIdPartAsString.charAt(0) + PATH_DELIMITER_CHARACTER +
+            blobIdPartAsString.charAt(1) + PATH_DELIMITER_CHARACTER +
+            blobIdPartAsString.charAt(2) + PATH_DELIMITER_CHARACTER +
+            blobIdPartAsString.charAt(3) + PATH_DELIMITER_CHARACTER +
+            blobIdPartAsString.substring(4);
```

Comment:
No it was intended NOT to be in asString in prder to preserve blobId::asString stability.

With your code:

```
MinIOGenerationAwareBlobId blobId;

```
blobId.asString 

!= blobIdFactory.parse(blobId.asString).asString
```